test/test_torch.py

```diff
@@ -5404,6 +5404,229 @@ def test_narrow_empty(self, device):
+            ((10, 10), 'constant', ((2, 2), (2, 2)), {}),
+            ((10, 10), 'constant', ((2, 2), (2, 2)), {'constant_values': 10}),
+            ((5, 2, 4, 1, 7), 'constant', ((1, 2), (6, 4), (10, 7), (3, 3), (0, 0)), {'constant_values': 89}),
+            ((15,), 'constant', ((100, 100),), {'constant_values': 1234}),
+            ((15,), 'constant', ((0, 1000),), {'constant_values': 1234}),
+            ((15,), 'constant', ((100, 0),), {'constant_values': 1234}),
+            ((0,), 'constant', ((100, 0),), {'constant_values': 1234}),
+            ((0,), 'constant', ((0, 100),), {'constant_values': 1234}),
+            ((0,), 'constant', ((0, 0),), {'constant_values': 1234}),
+            ((0, 0), 'constant', ((100, 0), (100, 100)), {'constant_values': 1234}),
+            ((0, 0), 'constant', ((1, 0), (1, 0)), {'constant_values': 1234}),
+            ((0, 0), 'constant', ((0, 100), (1, 0)), {'constant_values': 1234}),
+            ((0, 0), 'constant', ((1, 1), (1, 1)), {'constant_values': 1234}),
+            ((0, 0), 'constant', ((0, 0), (0, 0)), {'constant_values': 1234}),
+
+            ((10, 10, 10), 'empty', ((0, 0), (0, 0), (0, 0)), {}),
+            ((4, 13, 2, 5), 'empty', ((3, 2), (8, 10), (1, 0), (5, 6)), {}),
+            ((10, 10), 'empty', ((2, 2), (2, 2)), {}),
+            ((10, 10), 'empty', ((2, 2), (2, 2)), {}),
+            ((5, 2, 4, 1, 7), 'empty', ((1, 2), (6, 4), (10, 7), (3, 3), (0, 0)), {}),
+            ((15,), 'empty', ((100, 100),), {}),
+            ((15,), 'empty', ((0, 1000),), {}),
+            ((15,), 'empty', ((100, 0),), {}),
+            ((0,), 'empty', ((100, 0),), {}),
+            ((0,), 'empty', ((0, 100),), {}),
+            ((0,), 'empty', ((0, 0),), {}),
+            ((0, 0), 'empty', ((100, 0), (100, 100)), {}),
+            ((0, 0), 'empty', ((1, 0), (1, 0)), {}),
+            ((0, 0), 'empty', ((0, 100), (1, 0)), {}),
+            ((0, 0), 'empty', ((1, 1), (1, 1)), {}),
+            ((0, 0), 'empty', ((0, 0), (0, 0)), {}),
+        ]
+
+        for input_size, mode, pad_width, kwargs in test_cases:
+            input = make_tensor(input_size, device, dtype, low=-9, high=9)
+            input_np = input.cpu().numpy()
+
+            result = input.pad(pad_width, mode, **kwargs).cpu()
+            result_np = torch.tensor(np.pad(input_np, pad_width, mode, **kwargs))
+
+            msg = f'input_size: {input_size}, mode: {mode}, pad_width: {pad_width}, kwargs: {kwargs}'
+
+            self.assertEqual(result.size(), result_np.size(), msg=msg)
+
+            if mode == 'empty':
+                # For empty mode, only check the non-padded elements,
+                # since the padded elements could have any value
+
+                if input.numel() == 0:
+                    continue
+
+                slices = []
+
+                pw = torch.tensor(pad_width)
+
+                if pw.numel() == 1:
```

Comment:
One idea that might make this part of the test simpler and more maintainable is to pad the original input with the same arguments but mode constant and a constant value of NaN, then add the results and then compare those tensors. This should make all the empty padded regions NaN (since NaN + anything is NaN). This doesn't test the pad results directly, however, and won't work on tensors that aren't floating point or complex (maybe empty testing could be skipped on them?).

Or maybe it's not worth trying to simplify this.

Reply:
True but assertEqual has an `equal_nan` kwarg just like isclose does:

https://pytorch.org/docs/master/generated/torch.isclose.html?highlight=isclose#torch.isclose

Reply:
No need to change this for this PR, but maybe an idea for the future